src/ophyd_async/tango/demo/_detector.py

```diff
@@ -0,0 +1,40 @@
+import asyncio
+
+from ophyd_async.core import (
+    AsyncStatus,
+    DeviceVector,
+)
+from ophyd_async.tango import TangoReadable
+
+from ._counter import TangoCounter
+from ._mover import TangoMover
+
+
+class TangoDetector(TangoReadable):
+    counters: DeviceVector[TangoCounter]
+    mover: TangoMover
+
+    def __init__(self, *args, **kwargs):
+        if "counters_kwargs" in kwargs:
+            self._counters_kwargs = kwargs.pop("counters_kwargs")
+        if "mover_kwargs" in kwargs:
+            self._mover_kwargs = kwargs.pop("mover_kwargs")
```

Comment:
What are these kwargs doing?

I'm adding typing at the moment, and args and kwargs are making things difficult, so if we could remove them that would help...

Reply:
Those were there because tango devices need a trl to connect. I refactored the code so that devices can be initialized without a trl but one will have to be set before connecting.